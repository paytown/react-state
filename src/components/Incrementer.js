import React, { PureComponent } from 'react';


export default class Incrementer extends PureComponent {
  state = {
    count: 0
  };

  counter = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  }

  reset = () => {
    this.setState(() => {
      return { count: 0 };
    });
  }

  render() {
    const { count } = this.state;
    return (
      <>
        <p>You pushed that button {count} times!</p>
        <button onClick={this.counter} style={{ cursor: 'pointer' }}>Poke!</button>
        <button id="reset" onClick={this.reset} style={{ cursor: 'pointer' }}>Reset</button>
      </>
    );
  }
}
