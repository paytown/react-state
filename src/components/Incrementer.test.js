import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('incrememnter', () => {
  it('adds a number when button clicked', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('increments on click', () => {
    const wrapper = shallow(<Incrementer />);
    wrapper.find('button').at(0).simulate('click');

    expect(wrapper.state('count')).toEqual(1);
  });
});
