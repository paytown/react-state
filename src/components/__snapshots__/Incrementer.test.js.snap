// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrememnter adds a number when button clicked 1`] = `
<Fragment>
  <p>
    You pushed that button 
    0
     times!
  </p>
  <button
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    Poke!
  </button>
  <button
    id="reset"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    Reset
  </button>
</Fragment>
`;
